### ISSUE TITLE
Multi-arch builds

### ISSUE THREAD

**Issue by cedricve** [balena-io/deploy-to-balena-action#200]
We are running a multi arch build in a github workflow using `docker buildx build`, and are wondering how to manage the `balena deploy` so all architecture images are merged in a single release.

https://github.com/kerberos-io/agent/blob/master/.github/workflows/docker.yml#L63-L67

**Comment by pipex** [comment]
Hi @cedricve, thanks for your question. I was also looking into this myself. Unfortunately the limitation here is that the [balena CLI](https://github.com/balena-io/balena-cli/), which this action uses under the hood, does not yet support pushing multiple images for the same release to our registry, and it doesn't push the multi-arch manifest either. Multi-arch support is on our roadmap but we don't have an ETA for that feature yet or the implementation details for it.

That being said, I am looking into adding a `useDeploy` feature, that could use image artifacts from a previous step in order to do a [balena deploy](https://www.balena.io/docs/reference/balena-cli/#deploy-fleet-image) instead of a `balena push`

The action would look in its local image store for one or more `<projectName>_<serviceName>:<architecture>` or  `<projectName>_<serviceName>:latest` images (loaded by a previous step of the same job), and then deploy the images matching the target fleet architecture.

What do you think about that as an interface?

**Comment by cedricve** [comment]
Thanks @pipex  that would already make more sense. Where I think a lot of confusion starts is the difference between the "direct" deployment on Balena Hub.

1) When browsing to the Balena Hub, opening the Kerberos Agent project, and pressing the **Get Started** button, it will list all the available supported architectures. However under the hood the fleet was pointed to the arm7 architecture, so only that image is available in the balena registry. If you select an ARM64 device for example Raspberry Pi 4 from the list, it will not boot. Which is not expected and can been seen by the users as malfunctioning of Balena.

![Screenshot 2022-09-27 at 08 21 55](https://user-images.githubusercontent.com/1546779/192448679-7f81e79a-b89e-43c9-9757-0b744753a346.png)

![Screenshot 2022-09-27 at 08 22 07](https://user-images.githubusercontent.com/1546779/192448688-a1403122-078f-4dc1-ab24-88739f60e8db.png)

2. On the other hand when you "fork" the fleet to your own fleet. You can select all (even non-supported) architectures from the list. This helps to get the other "available" architectures which are supported, as after creation it will make a new release using the right base image. However, here you can select a non-supported architecture from the list, and then it will not work.

![Screenshot 2022-09-27 at 08 26 20](https://user-images.githubusercontent.com/1546779/192449294-49558598-e94c-49a4-b949-f37a2756abfc.png)

Bottom line is that it's really not straight forward to a users what is working and what is not.

**Comment by cedricve** [comment]
Thanks for explaining, I didn't know that v8 was backwards compatible. However that wouldn't solve the amd64 boards right, as the target board on your fleet is determining the specific architecture build.

**Comment by cedricve** [comment]
Understood! So we’ll need to wait for the multi architecture capability. I think it would be nice and intuitive you can just select the right board (and under the hood the correct base image is selected arm or amd).

**Comment by klutchell** [comment]
I'm hoping to have more to share on multi-arch apps soon!

**Comment by cedricve** [comment]
exciting !

**Comment by klutchell** [comment]
Hey @avihaibd, thanks for your interest in this feature! Here is the related roadmap item that you can follow for updates!

https://roadmap.balena.io/posts/28/mixed-architecture-fleets-and-apps

In short, there are a lot of backend changes required to our API and database to support mixed device fleets, and multi-arch applications, but it is a priority project that we are excited to progress!

**Comment by robsonvn** [comment]
Meanwhile, this is what I am doing

```Dockerfile.template
FROM arm64v8/nginx:1.25 as nginx-aarch64
FROM arm32v7/nginx:1.25 as nginx-armv7hf
FROM nginx:1.25 as nginx-amd64

FROM nginx-%%BALENA_ARCH%%
```
